src/main/scala-2.12/org/ergoplatform/appkit/ergotool/dex/CancelOrderCmd.scala

```diff
@@ -147,18 +155,23 @@ object CancelOrder {
           .getOrElse(sys.error(s"cannot extract seller PK from order box $orderBoxId"))
         require(sellerPk == recipientAddress.getPublicKey,
           s"sell order box $orderBoxId can be claimed with $sellerPk PK, while your's is ${recipientAddress.getPublicKey}")
-        OutBoxProto(orderBox.getValue - txFee, Seq(orderBox.getTokens.get(0)), outboxContract)
+        OutBoxProto(orderBox.getValue - txFee, Seq(orderBox.getTokens.get(0)), Seq(), outboxContract)
       } else if (util.Arrays.equals(orderBoxContractTemplate.getBytes, BuyerContract.contractTemplate.getBytes)) {
         // buy order
         val buyerPk = BuyerContract.buyerPkFromTree(orderBox.getErgoTree)
           .getOrElse(sys.error(s"cannot extract buyer PK from order box $orderBoxId"))
         require(buyerPk == recipientAddress.getPublicKey,
           s"buy order box $orderBoxId can be claimed with ${buyerPk} PK, while yours is ${recipientAddress.getPublicKey}")
         // as a workaround for https://github.com/ScorexFoundation/sigmastate-interpreter/issues/628
-        // box.tokens cannot be empty
-        val token = new ErgoToken("21f84cf457802e66fb5930fb5d45fbe955933dc16a72089bf8980797f24e2fa1",
-          0L)
-        OutBoxProto(orderBox.getValue - txFee, Seq(token), outboxContract)
+        // box.tokens cannot be empty, so we mint new token
+        val token = new ErgoToken(orderBox.getId, 1L)
+        // fill R4, R5, R6 according to https://github.com/ergoplatform/eips/blob/master/eip-0004.md
+        val registers: Seq[ErgoValue[_]] = Seq(
+          ErgoValue.of("CANCELDEXBUY".getBytes), // token name in R4 (see EIP-4)
+          ErgoValue.of("New token each time DEX buy order is cancelled".getBytes), // token description in R5 (see EIP-4)
+          ErgoValue.of("2".getBytes), // number of decimals in R6 (see EIP-4)
+        )
+        OutBoxProto(orderBox.getValue - txFee, Seq(token), registers, outboxContract)
```

Comment:
This code can be incapsulated into a method which adds a new minted token to the given box builder using given parameters.

Reply:
Good call! I added it in https://github.com/aslesarenko/ergo-appkit/pull/34 and use here.